SIL.Windows.Forms.WritingSystems/LanguageLookupModel.cs

```diff
@@ -67,11 +67,23 @@ public IEnumerable<LanguageInfo> MatchingLanguages
 					yield break;
 				}
 
-				foreach (LanguageInfo li in _languageLookup.SuggestLanguages(_searchText).Where(li => MatchingLanguageFilter == null || MatchingLanguageFilter(li)))
+				foreach (LanguageInfo li in _languageLookup.SuggestLanguages(_searchText).Where(li => (MatchingLanguageFilter == null || MatchingLanguageFilter(li)) && ShowDialectsFilter(li)))
 					yield return li;
 			}
 		}
 
+		private bool ShowDialectsFilter(LanguageInfo li)
```

Comment:
The "Show" verb here had me expecting that it was a command you give to enable showing of dialects. Also, this is specifically about region codes, so how about using RegionCodesFilter().